bundles/org.openhab.binding.bluetooth.daikinmadoka/src/main/java/org/openhab/binding/bluetooth/daikinmadoka/handler/DaikinMadokaHandler.java

```diff
@@ -0,0 +1,782 @@
+        if (!characteristic.getUuid().equals(DaikinMadokaBindingConstants.CHAR_NOTIF_UUID)) {
+            return;
+        }
+
+        // A message cannot be null or have a 0-byte length
+        byte[] msgBytes = characteristic.getByteValue();
+        if (msgBytes == null || msgBytes.length == 0) {
+            return;
+        }
+
+        this.uartProcessor.chunkReceived(msgBytes);
+    }
+
+    private void submitCommand(BRC1HCommand command) {
+        Executor executor = commandExecutor;
+
+        if (executor != null) {
+            executor.execute(() -> processCommand(command));
+        }
+    }
+
+    private void processCommand(BRC1HCommand command) {
+        logger.debug("[{}] ProcessCommand {}", super.thing.getUID().getId(), command.getClass().getSimpleName());
+
+        try {
+            currentCommand = command;
+            uartProcessor.abandon();
+
+            if (device == null || device.getConnectionState() != ConnectionState.CONNECTED) {
+                logger.debug("Unable to send command {} to device {}: not connected",
+                        command.getClass().getSimpleName(), address);
+                command.setState(BRC1HCommand.State.FAILED);
+                return;
+            }
+
+            if (!resolved) {
+                logger.debug("Unable to send command {} to device {}: services not resolved",
+                        command.getClass().getSimpleName(), device.getAddress());
+                command.setState(BRC1HCommand.State.FAILED);
+                return;
+            }
+
+            BluetoothCharacteristic charWrite = device
+                    .getCharacteristic(DaikinMadokaBindingConstants.CHAR_WRITE_WITHOUT_RESPONSE_UUID);
+            if (charWrite == null) {
+                logger.warn("Unable to execute {}. Characteristic '{}' could not be found.",
+                        command.getClass().getSimpleName(),
+                        DaikinMadokaBindingConstants.CHAR_WRITE_WITHOUT_RESPONSE_UUID);
+                command.setState(BRC1HCommand.State.FAILED);
+                return;
+            }
+
+            BluetoothCharacteristic charNotif = this.device
+                    .getCharacteristic(DaikinMadokaBindingConstants.CHAR_NOTIF_UUID);
+
+            if (charNotif != null) {
+                device.enableNotifications(charNotif);
+            }
+
+            charWrite.setValue(command.getRequest());
+            command.setState(BRC1HCommand.State.ENQUEUED);
+            device.writeCharacteristic(charWrite);
+
+            if (this.config != null) {
+                if (!command.awaitStateChange(this.config.commandTimeout, TimeUnit.MILLISECONDS,
+                        BRC1HCommand.State.SUCCEEDED, BRC1HCommand.State.FAILED)) {
+                    logger.debug("Command {} to device {} timed out", command, device.getAddress());
+                    command.setState(BRC1HCommand.State.FAILED);
+                }
+            }
+
+        } catch (Exception e) {
```

Comment:
The idea here is I need to make the command terminate (execute the finally block), whatever happens. Any suggestion to achieve this goal ?

Reply:
The finally does not look that important to me.
Looking at it, I would rather suggest to not have `currentCommand` as a field in the class. Its only purpose seems to be to have a handle for interrupting it in case of a `dispose` call. But for this it would be cleaner to simply shut down the executor and tell it to interrupt existing jobs - so you could cancel the `currentCommand` within the `InterruptedException` catch clause.

Reply:
I still need the "currentCommand" to track it somewhere else, so I implemented a trick to let the exception bubble up.